power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/serializer.hpp

```diff
@@ -59,6 +63,147 @@ MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS) {
 
 namespace power_grid_model::meta_data {
 
+namespace json_converter {
+
+struct MapArray {
+    MapArray(uint32_t size_input) : size{size_input}, empty{size_input == 0} {}
+
+    uint32_t size;
+    bool empty;
+    bool begin{true};
+};
+
+struct JsonConverter : msgpack::null_visitor {
+    static constexpr char sep_char = ' ';
+
+    Idx indent;
+    Idx max_indent_level;
+    std::stringstream ss{};
+    std::stack<MapArray> map_array{};
+
+    void print_indent() {
+        if (indent < 0) {
+            return;
+        }
+        Idx const indent_level = static_cast<Idx>(map_array.size());
+        if (indent_level > max_indent_level) {
+            if (map_array.top().begin) {
+                map_array.top().begin = false;
+                return;
+            }
+            ss << sep_char;
+            return;
+        }
+        ss << '\n';
+        Idx indent_size = indent_level * indent;
+        while (indent_size-- != 0) {
+            ss << sep_char;
+        }
+    }
+
+    void print_key_val_sep() {
+        if (indent < 0) {
+            return;
+        }
+        ss << sep_char;
+    }
+
+    bool visit_nil() {
+        ss << "null";
+        return true;
+    }
+    bool visit_boolean(bool v) {
+        if (v) {
+            ss << "true";
+        } else {
+            ss << "false";
+        }
+        return true;
+    }
+    bool visit_positive_integer(uint64_t v) {
+        ss << v;
+        return true;
+    }
+    bool visit_negative_integer(int64_t v) {
+        ss << v;
+        return true;
+    }
+    bool visit_float32(float v) { return visit_float64(v); }
+    bool visit_float64(double v) {
+        if (std::isinf(v)) {
+            if (v > 0.0) {
+                ss << '"' << "inf" << '"';
+            } else {
+                ss << '"' << "-inf" << '"';
+            }
```

Comment:
why not just `'"inf"'`?
```suggestion
            if (v > 0.0) {
                ss << '"inf"';
            } else {
                ss << '"-inf"';
            }
```

if anything, i would just do

```suggestion
            ss << '"' << (v > 0.0 ? "" : "-") << "inf" << '"';
```

or even

```suggestion
            ss << (v > 0.0 ? '"inf"' : '"-inf"');
```

or

```suggestion
            auto const str = v > 0.0 ? "inf"sv: "-inf"sv;
            visit_str(str.c_str(), str.size());
```